examples/cord_epsim/main.c

```diff
@@ -67,33 +69,51 @@ static gcoap_listener_t listener = {
 
 int main(void)
 {
+    char addrstr[IPV6_ADDR_MAX_STR_LEN];
     puts("Simplified CoRE RD registration example\n");
 
     /* fill riot info */
     sprintf(riot_info, "{\"ep\":\"%s\",\"lt\":%u}",
             cord_common_get_ep(), CORD_LT);
 
     /* parse RD address information */
-    sock_udp_ep_t rd_ep = {
-        .family    = AF_INET6,
-        .netif     = SOCK_ADDR_ANY_NETIF,
-        .port      = RD_PORT,
-    };
-
-    /* parse RD server address */
-    if (ipv6_addr_from_str((ipv6_addr_t *)&rd_ep.addr.ipv6, RD_ADDR) == NULL) {
+    sock_udp_ep_t rd_ep;
+
+    if (sock_udp_str2ep(&rd_ep, RD_ADDR) < 0) {
         puts("error: unable to parse RD address from RD_ADDR variable");
         return 1;
     }
 
+    /* if netif is not specified in addr and it's link local */
+    if ((rd_ep.netif == SOCK_ADDR_ANY_NETIF) &&
+         ipv6_addr_is_link_local((ipv6_addr_t *) &rd_ep.addr.ipv6)) {
+        /* if there is only one interface we use that */
+        if (gnrc_netif_numof() == 1) {
+            rd_ep.netif = (uint16_t)gnrc_netif_iter(NULL)->pid;
+        }
+        /* if there are many it's an error */
+        else {
+            puts("error: must specify an interface for a link local address");
+            return 1;
+        }
+    }
+
+    if (rd_ep.port == 0) {
+        rd_ep.port = COAP_PORT;
+    }
+
+    ipv6_addr_to_str(addrstr,
+                     (ipv6_addr_t *) &rd_ep.addr.ipv6,
+                     sizeof(addrstr));
+
     /* register resource handlers with gcoap */
     gcoap_register_listener(&listener);
 
     /* print RD client information */
     puts("epsim configuration:");
     printf("         ep: %s\n", cord_common_get_ep());
     printf("         lt: %is\n", (int)CORD_LT);
-    printf(" RD address: [%s]:%u\n\n", RD_ADDR, (unsigned)RD_PORT);
+    printf(" RD address: [%s]:%u\n\n", addrstr, rd_ep.port);
```

Comment:
Would be good to also print the interface here?

Reply:
Note to self: need to adapt shell parser for release tests ;-) https://github.com/RIOT-OS/Release-Specs/blob/d7784817b99fe16622f63e741de641f7d81acac2/09-coap/test_spec09.py#L36-L37

Reply:
Changed

Reply:
[Done](https://github.com/RIOT-OS/Release-Specs/pull/168/commits/ffb51537c6547027f5e74b01cba13b6b5708661f) so it works with both versions (at least for this release cycle this is required, as the release branch does not print it as such)

Reply:
Or do you plan to backport this?

Reply:
I don't know if the change is really so important to backport it right now that we are testing RC2

Reply:
Given that there is an automated test case involving this application, this would be really easy to test in a potential RC3, but agreed.